ffi/dart/pubspec.yaml

```
@@ -3,14 +3,15 @@ version: 0.0.0
 repository: https://github.com/unicode-org/icu4x
 
 environment:
-  sdk: ^3.4.0-204.0.dev
+  sdk: ^3.7.0
```

```suggestion
  sdk: '>=3.7.0 <3.8.0'
```

---

That would be a very tight constraint, and require more frequent updating. I would suggest `>=3.7.0 <4.0.0` if you are worried about breaking changes.

---

yes it is tight, because it's an experimental feature. it broke between 3.4 and 3.7, it will break between 3.7 and 3.12

---

And it will also break between 3.7.1 and 3.7.2 - so I don't see a reason to arbitrarily cut off here. And pinning it completely would make the usability even worse.

---

Ok then this should be `=3.7.0`, which is how `^3.4.0-204.0.dev` behaved iiuc

---

It's fine for my use case, as I don't use the package but just copy over files, but this makes the package unusable and I strongly recommend against it. If we go that route, I would add a comment to the readme that this package is not meant for usage, and any user should fork it and use that fork instead.

I suggest either using `^3.7.0` or the more conservative `>=3.7.0 <4.0.0`. Which is also why I advise against pinning the CI Dart version, as this will just hide failures, not make them go away.

---

I consider this package unusable because it uses unreleased compiler features. *If* you want to use it, we guarantee it works for 3.7.0, but nothing else.

---

It's standard practice for most published Dart packages, and all published by the Dart team, to update the minimum version to the current stable as far as possible, and test against stable and dev versions. Users with older SDK versions can use older published versions.
For more experimental packages, you might require an even newer SDK, and only test against dev, as the expectation is that users trying out experimental packages will have the newest SDK versions. As this package is not yet published, you don't make any guarantees to it's usability, so any SDK bound and testing is fine. The bounds and testing procedures set now are only meant to support us in development. I find it easiest to set loose bounds, so that we don't have to go through the hassle of updating the bounds often while in development. As soon as the package is published, we can go to the established practice of keeping it up to date with the newest stable version.

---

We're not a Dart development team. We just want CI to work, and getting an error that you have the wrong Dart SDK is a million times better for us than getting a test failure in a language few people on the team have ever used.

---

Then I would suggest deleting the testing from CI, so it always passes. This would be preferable to having a package which passes tests, but cannot be used. Would that be a compromise?

If this is not acceptable, then I will have to fork this package for further use, which would mean an additional workload I would rather not have.

---

But we are testing across changes to ICU4X FFI. We are just not testing across changes in the Dart compiler.

I thought you are using your own pubspec?